### PR TITLE
[MIG] medical_pathology: Upgrade to v10

### PR DESCRIPTION
**Pull request by lasley**
Migrate `medical_pathology` to v10. 

This also includes the license audit for #183

### CODE REVIEW

File: medical_pathology/models/medical_pathology_group.py
Comment (lasley) [inline medical_pathology/models/medical_pathology_group.py]: I'm actually going to delete this model entirely unless anyone cares.

Its intent was to provide a categorization for pathologies that are not within a standardized coding - basically internal/external categories. 

This has proven too complex, creates confusion, and so far nobody (in terms of our customers) is even using it.

File: medical_pathology/models/medical_pathology.py
Comment (lasley) [inline medical_pathology/models/medical_pathology.py]: `category_ids` - A pathology should be able to relate to multiple category types. It should, however, verify unique `category.code_type_id` so that only one code type per category is introduced

Comment (lasley) [inline medical_pathology/models/medical_pathology.py]: Note: this is so that a pathology can be related to all of its different code types (ICD-9, 10, SNOMED).

There is one hole here though - each of the code types contains slightly different naming and descriptions, while still being related to the same pathology. I am not quite sure how to circumvent this & will be adding it to v11 roadmap

File: medical_pathology/README.rst
Comment (obulkin) [inline medical_pathology/README.rst]: s/ths/this

File: medical_pathology/README.rst
Comment (obulkin) [inline medical_pathology/README.rst]: Just noting here that this line will no longer be relevant if the group model is removed as per https://github.com/OCA/vertical-medical/pull/190#discussion_r132572522. The same is true for the images below, which will need to be updated to show the new menu structure.

File: medical_pathology/__openerp__.py
Comment (obulkin) [inline medical_pathology/__openerp__.py]: This file shouldn't be here.

File: medical_pathology/demo/medical_pathology_category_demo.xml
Comment (obulkin) [inline medical_pathology/demo/medical_pathology_category_demo.xml:7]: All these IDs and the ones in ``medical_pathology_demo.xml`` need to include ``demo`` to align with OCA standards: https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#demo-records

File: medical_pathology/models/medical_pathology.py
Comment (obulkin) [inline medical_pathology/models/medical_pathology.py]: The error message here needs to be reworded since you can actually create a pathology that is both an ancestor and a descendant so long as this doesn't introduce a loop in the pathology hierarchy.

File: medical_pathology/models/medical_pathology_category.py
Comment (obulkin) [inline medical_pathology/models/medical_pathology_category.py]: There are lots of ``openerp`` imports throughout that need to change to ``odoo``.

File: medical_pathology/tests/test_medical_pathology.py
Comment (obulkin) [inline medical_pathology/tests/test_medical_pathology.py]: Minor point but there's no reason to do this in ``setUp`` as the record is only used in one of the two test cases.

**Review by obulkin — CHANGES_REQUESTED** [review]

File: medical_pathology/demo/medical_pathology_category_demo.xml
Comment (lasley) [reply medical_pathology/demo/medical_pathology_category_demo.xml:7]: I'm going to hold off on this one - it causes cascading issues in our incubator & I've already resolved them once this version. We'll hold off till next version.

File: medical_pathology/models/medical_pathology.py
Comment (lasley) [reply medical_pathology/models/medical_pathology.py]: How so? If a pathology is both a descendant (child) and an ancestor (parent), there is a loop:

```
pathology_a.child_ids = [(4, pathology_b.id)]
pathology_c.parent_id = pathology_b.id
pathology_a.parent_id = pathology_c.id
```
Which is:
```
B => C => A => B
```

Comment (lasley) [reply medical_pathology/models/medical_pathology.py]: Oh I see - we're not actually testing for that. Ok yeah I'd rather just keep the tortoise/hair here instead of a custom implementation. Changing verbiage

File: medical_pathology/tests/test_medical_pathology.py
Comment (lasley) [reply medical_pathology/tests/test_medical_pathology.py]: Hmmm yeah but it's 2 of 2, which is 100% of the test cases. Seems worth it IMO - saves 3 lines

**Comment by lasley** [comment]
Thanks for the review @obulkin - all but a few of the changes made. I also forgot to remove the genetics stuff on my first pass, so did that now. We have a planned genetics module that will introduce the proper fields here - adding them as character fields is not helpful and will impede that module.

File: medical_pathology/demo/medical_pathology_category_demo.xml
Comment (woodbrettm) [reply medical_pathology/demo/medical_pathology_category_demo.xml:7]: Perhaps we should also specify a proper demo id naming scheme as well for next version. I think the general agreement was demo_model_name_#, but don't think it's in docs yet.

Perhaps also documentation for if a demo record has a name field, what to name it.

Comment (lasley) [reply medical_pathology/demo/medical_pathology_category_demo.xml:7]: I don't think it's that much of an issue - it's demo data.

Comment (lasley) [reply medical_pathology/demo/medical_pathology_category_demo.xml:7]: But then again that's probably because I'm also the bean counter. Regardless, we'll nail this in one pass at some point in a ticket specifically for it.

Comment (lasley) [reply medical_pathology/demo/medical_pathology_category_demo.xml:7]: Keep in mind that major restructuring is planned for v11 too (well ok maybe not major, but a lot). We're going to invalidate a lot of stuff

File: medical_pathology/tests/test_medical_pathology.py
Comment (obulkin) [reply medical_pathology/tests/test_medical_pathology.py]: The second test case doesn't use this attribute, so it's 1 of 2. In any event, it's a pretty minor improvement and doesn't really need to happen.

File: medical_pathology/demo/medical_pathology_category_demo.xml
Comment (obulkin) [reply medical_pathology/demo/medical_pathology_category_demo.xml:7]: @bmw95 - Actually the current approach outlined in the guidelines is to put ``demo`` at the very end of the ID, but this is fairly arbitrary and putting it at the very beginning would probably work just as well in 99% of cases.

@lasley - Got it. Fair enough.

File: medical_pathology/tests/test_medical_pathology.py
Comment (lasley) [reply medical_pathology/tests/test_medical_pathology.py]: Ah yup you're right. Done!

**Comment by obulkin** [comment]
Most of my comments have been addressed now, but the README still uses old screenshots that show a ``Pathology Group`` menu item.

**Comment by lasley** [comment]
Ahhh stupid screenshots - done.

**Review by obulkin — APPROVED** [review]
Haha. Thanks, @lasley. All my comments have been addressed, so the PR should be ready for merge.

**Comment by lasley** [comment]
cc @jbeficent @rgarnau

File: medical_pathology/__manifest__.py
Comment (rgarnau) [inline medical_pathology/__manifest__.py]: In here it should be: "https://github.com/OCA/vertical-medical" as mentioned in https://github.com/OCA/maintainer-tools/blob/4679ce184af4d2af3175d0b52f3178b0c231b83c/template/module/__openerp__.py#L9

**Review by rgarnau — COMMENTED** [review]
The only thing I would suggest is that in 'Pathology Categories' in 'Parent Category' all the preceding categories would appear in the name as ".../Category 2/Category 1". However, for the tree view it could get too long.

**Comment by lasley** [comment]
@rgarnau - thanks for the review, sorry i missed it. I agree that it would be nice to show the parents, but yeah I'm not sure how it would look in a lot of contexts. Seeing as we'll probably be redesigning this in v11 anyways, it's probably best to figure it out then. Do you agree?

**Review by rgarnau — APPROVED** [review]
:+1:

**Pull request merged**
